openff/evaluator/backends/backends.py

```diff
@@ -92,6 +92,7 @@ def __init__(
 
         if self._number_of_gpus > 0:
             assert self._preferred_gpu_toolkit is not None
+            assert self._number_of_gpus == 1, f'only 1 gpu per worker is supported'
```

Comment:
I'm not sure this is an improvement; even if multiple GPUs per worker is not common, I'd like to avoid locking out future use cases

Reply:
I agree with @mattwthompson on this, there may be some use cases in the future.

Reply:
I added this to simplify the gpu-id calculation in the round-robin gpu assignment stage.

yes, it can be removed.